src/rules/selector-pseudo-class-blacklist/__tests__/index.js

```diff
@@ -0,0 +1,117 @@
+import {
+  messages,
+  ruleName,
+} from ".."
+import rules from "../../../rules"
+import { testRule } from "../../../testUtils"
+
+const rule = rules[ruleName]
+
+testRule(rule, {
+  ruleName,
+  config: [[
+    "focus",
+    "global",
+    "input-placeholder",
+    "not",
+    "nth-last-child",
+  ]],
+  skipBasicChecks: true,
+
+  accept: [ {
+    code: "a {}",
+  }, {
+    code: "a:hover {}",
+  }, {
+    code: "a:nth-child(5) {}",
+  }, {
+    code: ":root {}",
+  }, {
+    code: "a:HOVER {}",
+  }, {
+    code: "a:hover, a:nth-child(5) {}",
+  }, {
+    code: "a::before {}",
+  }, {
+    code: "a:nth-child(5)::before {}",
+  }, {
+    code: "a:-moz-placeholder {}",
+  }, {
+    code: "a:-MOZ-PLACEholder {}",
+  } ],
+
+  reject: [ {
+    code: "a:focus {}",
+    message: messages.rejected("focus"),
+    line: 1,
+    column: 2,
+  }, {
+    code: "div:nth-LAST-child {}",
+    message: messages.rejected("nth-LAST-child"),
+    line: 1,
+    column: 4,
+  }, {
+    code: "a,\n:global {}",
+    message: messages.rejected("global"),
+    line: 2,
+    column: 1,
+  }, {
+    code: "input:-ms-input-placeholder {}",
+    message: messages.rejected("-ms-input-placeholder"),
+    line: 1,
+    column: 6,
+  }, {
+    code: "input:-Ms-INPUT-placeholder {}",
+    message: messages.rejected("-Ms-INPUT-placeholder"),
+    line: 1,
+    column: 6,
+  }, {
+    code: "a:not(::selection) {}",
+    message: messages.rejected("not"),
+    line: 1,
+    column: 2,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [[
+    "/^last/",
+  ]],
+  skipBasicChecks: true,
+
+  accept: [ {
+    code: "a {}",
+  }, {
+    code: "a:first-child() {}",
+  }, {
+    code: "a:nth-LAST-child(5) {}",
+  } ],
+
+  reject: [ {
+    code: "a:last-child {}",
+    message: messages.rejected("last-child"),
+    line: 1,
+    column: 2,
+  }, {
+    code: "a:last-of-child {}",
+    message: messages.rejected("last-of-child"),
+    line: 1,
+    column: 2,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: ["variable"],
```

Comment:
The white/blacklist rules accept a string for their primary option, so this valid. I can change it to a single item array if you like though?